Change update from put to patch

zancas opened the pull request:
@caphrim007 
Please reassign to @pjbreaux , and @wojtek0806 , after review comments...   I want to make sure all core developers look at this before merge.
Issues:
Fixes #578 #564

Problem: The use of PUT in update was causing unintended side-effects.
This is costly behavior..  we probably shouldn't force the user to
submit more data than they intend!

Analysis: I've changed "put" to "patch" in the update method.
This seems like a major change, so I'd appreciate lots of feedback.

Among other things I'd like to validate that the f5-openstack-agent
consumer does _not_ break against this new version.

Tests:
- f5/bigip/test/test_resource.py
- f5/bigip/tm/sys/test/test_sys_application.py

caphrim007 committed:
Fixes the skipping of ltm pool tests

Issues:
Fixes #508

Problem:
The pytest.skip decorator was being used in a method of one of the test classes
and this was causing all of the tests in the file to be skipped.

Analysis:
The correct syntax for this is to use "pytest.mark.skip", but I took a look
and it turns out that these tests work just fine. So I removed the skip
entirely

Tests:
  * test/functional/tm/ltm/test_pool.py

zancas committed:
Change update from put to patch

Issues:
Fixes #578

Problem: The use of PUT in update was causing unintended side-effects.
This is costly behavior..  we probably shouldn't force the user to
submit more data than they intend!

Analysis: I've changed "put" to "patch" in the update method.
This seems like a major change, so I'd appreciate lots of feedback.

Among other things I'd like to validate that the f5-openstack-agent
consumer does _not_ break against this new version.

Tests:

* f5/bigip/test/test_resource.py

* f5/bigip/tm/sys/test/test_sys_application.py

zancas commented:
Reviewers please consider this:

http://restful-api-design.readthedocs.io/en/latest/methods.html

(If you're not already familiar.)

zancas commented:
https://tools.ietf.org/html/rfc5789

caphrim007 commented:
i think people's intention is probably more patchy that puty. Curious to see if all the tests will pass.

zancas commented:
@caphrim007 all tests have indeed passed.

zancas commented:
@pjbreaux I'd like your input on this.

zancas commented:
I'd like @wojtek0806 's input, but he's away, so I'm OK merging without it.  I've talked with @jlongstaf about this issue and he thinks put --> patch is probably a good idea.

zancas committed:
Merge pull request #509 from caphrim007/fix-tests-for-ltm

Fixes the skipping of ltm pool tests

zancas committed:
The skipped tests of #509 had aliases

Issues:
Fixes #509

Problem: test_pool.py was being skipped.  Tests were failing because
of aliasing.   I've fixed the aliasing issues after applying @caphrim007s
removal of the skip.

Analysis: These tests should be run!

Tests: test/functional/tm/ltm/test_pool.py

zancas commented:
This PR is now "on top" of the bugfix.alias PR so that one should be closed first.

zancas commented:
@caphrim007 I looked at this build failure...  it looks like all of the tests pass...

caphrim007 commented:
@zancas going to rekick it to ensure that I have my infra changes correct

zancas commented:
OK @pjbreaux I've dropped this into your bailiwick.
